specs/AnimatedVisualPlayer Spec.md

```diff
@@ -0,0 +1,145 @@
+when animations should be destroyed and when they should be instantiated.
+This will be implemented using the previous new interfaces.
+
+# API Pages
+
+## AnimatedVisualPlayer.CacheMode property
+
+Gets or sets a value that determines how animations are cached. Defaults to 'Always'.
+
+The two modes for the property will specify the behavior the AnimatedVisualPlayer will have
+when the player is idle (when `PlayAsync` is not active).
+
+|Mode| Behavior|
+|---- | -------|
+|Always| `AnimatedVisualPlayer` will pre-create animations even before `PlayAsync()` is called,
+and not destroy any when player is idle. |
+|None | `AnimatedVisualPlayer` will not create animations until `PlayAsync()` is called,
+and will destroy them when it completes. |
+
+If you have an animation that will start on click, you may want to set `CacheMode` to `Always`
+when the mouse enters this control, 
+and back to `None` when mouse leaves to ensure that the animation starts immediately on click,
+but don't use up resources otherwise.
+
+Setting the `CacheMode` to `Always` when the mouse enters the control will ensure that
+the animation objects are preloaded so that the animation may start immediately when it needs to play. 
+
+If your player is does not have any animations to play and `CacheMode` is set to `Always`, the player will instantiate all animations immediately and play what was instantiated. 
+
+```c#
+private void Player_PointerEntered(object sender, PointerRoutedEventArgs e)
+{
+    myanimatedvisualplayer.CacheMode = AnimationsCacheMode.Always;
+}
+async private void Player_PointerExited(object sender, PointerRoutedEventArgs e)
+{              
+    myanimatedvisualplayer.CacheMode = AnimationsCacheMode.None;
+}
+
+async private void Player_OnClick(object sender, PointerRoutedEventArgs e)
+{              
+    myanimatedvisualplayer.PlayAsync(0.0, 1.0);
+}
+
+```
+
+If `PlayAsync` is called and the animations have not been instantiated due to `CacheMode` being set to `None`,
+`PlayAsync` will instantiate the animations so it can play the visual. 
+
+# API Details
+
+```c#
+namespace Microsoft.UI.Xaml.Controls
+{
+  unsealed runtimeclass AnimatedVisualPlayer
+  {
+      // Existing ...
+
+    //Sets the caching behavior of the AnimatedVisualPlayer
+    AnimationsCacheMode CacheMode;
+  }
+
+  public enum AnimationsCacheMode
+  {
+      Always = 0,
+      None
+  }
+}
+```
+
+## IAnimatedVisual2 Interface
+
+An extension to
+[IAnimatedVisual](https://docs.microsoft.com/uwp/api/Microsoft.UI.Xaml.Controls.IAnimatedVisual)
+that adds the ability to control when animation objects are created and destroyed.
+
+| Method | Meaning |
+|-------------|---------|
+| InstantiateAnimations | Instantiates the animation visuals for the provided progress value of the animation. |
+| DestroyAnimations | Destroys all animation visuals for a given animation.|
+
+```c# 
+interface IAnimatedVisual2 requires IAnimatedVisual
+{
+    void InstantiateAnimations(Double progressHint);
```

Comment:
Discussion: how should the `progressHint` be set? What does 'progress' mean?

"Progress" is a well-known Composition property name, shows up in the PlayAsync method too.

Is this parameter really necessary?

Recommend: remove parameter